image/postgres/scripts/docker-entrypoint.sh

```diff
@@ -84,18 +84,19 @@ docker_init_database_dir() {
 				NSS_WRAPPER_GROUP="$(mktemp)"
 				export LD_PRELOAD="$wrapper" NSS_WRAPPER_PASSWD NSS_WRAPPER_GROUP
 				local gid; gid="$(id -g)"
-				echo "postgres:x:$uid:$gid:PostgreSQL:$PGDATA:/bin/false" > "$NSS_WRAPPER_PASSWD"
-				echo "postgres:x:$gid:" > "$NSS_WRAPPER_GROUP"
+				printf 'postgres:x:%s:%s:PostgreSQL:%s:/bin/false\n' "$uid" "$gid" "$PGDATA" > "$NSS_WRAPPER_PASSWD"
+				printf 'postgres:x:%s:\n' "$gid" > "$NSS_WRAPPER_GROUP"
 				break
 			fi
 		done
 	fi
 
-	if [ -n "$POSTGRES_INITDB_WALDIR" ]; then
+	if [ -n "${POSTGRES_INITDB_WALDIR:-}" ]; then
 		set -- --waldir "$POSTGRES_INITDB_WALDIR" "$@"
 	fi
 
-	eval 'initdb --username="$POSTGRES_USER" --pwfile=<(echo "$POSTGRES_PASSWORD") --encoding=UTF8 '"$POSTGRES_INITDB_ARGS"' "$@"'
```

Comment:
I'm puzzled. Judging from this there was a point in time where `docker-library/postgres` was setting encoding explicitly. Yet in their repository history I can't find anything any reference to that.

Reply:
Yeah I think someone added that in when a customer was having issues with args in process indicators.  At the time they decided we only wanted to support UTF-8.  So it is possible that such a change could have a subtle impact that we wouldn't see until it hit the field.

Reply:
Good catch I didn't notice that. Here's the original PR from this repo: https://github.com/stackrox/stackrox/pull/5771

Perhaps I should re-add the encoding to make sure I don't mess with anything

Reply:
I re-added it to ensure this change doesn't have any unexpected side-effects